share/completions/gsettings.fish

```diff
@@ -0,0 +1,96 @@
+# Fish completions for glibs's "gsettings" configuration tool
+
+function __fish_complete_gsettings_args
+    set -l schema_commands get monitor writable range describe set reset reset-recursively list-keys list-children list-recursively
+    set -l key_commands get monitor writable range describe set reset
+
+    set -l schemadir
+    set -l offset 0
+    set -l cmd (commandline -poc)
+    set -e cmd[1]
+
+    if [ (count $cmd) -ge 2 ]; and string match -q -- '--schemadir' $cmd[1]
+        set schemadir --schemadir (echo $cmd[2])
+        set offset 2
+    end
+
+    if [ (count $cmd) -lt (math 1 + $offset) ]
+        return 1
+    end
+
+    set -l subcommand $cmd[(math 1 + $offset)]
+
+    if not contains -- $subcommand $schema_commands
+        return 1
+    end
+
+    if [ (count $cmd) -lt (math 2 + $offset) ]
+        # Non-relocatable schemas
+        gsettings $schemadir list-schemas ^/dev/null
+
+        # Relocatable schemas require a dconf path, but there is no universal way of
+        # finding the right paths to suggest here. Just default to '/'.
+        gsettings $schemadir list-relocatable-schemas ^/dev/null | while read -l line
+            echo $line:/
+        end
+        return 0
+    end
+
+    set -l schema $cmd[(math 2 + $offset)]
+
+    if not contains -- $subcommand $key_commands
+        return 1
+    end
+
+    if [ (count $cmd) -lt (math 3 + $offset) ]
+        gsettings $schemadir list-keys $schema ^/dev/null
+        return 0
+    end
+
+    if [ $subcommand != 'set' ]
+        return 1
+    end
+
+    set -l key $cmd[(math 3 + $offset)]
+
+    if [ (count $cmd) -lt (math 4 + $offset) ]
+        set -l key_type (gsettings $schemadir range $schema $key ^/dev/null | head -n 1)
```

Comment:
Thanks, that's a good suggestion. Usually there is just one line, except when there is an enum, in that case there is a (short) list of all possible values. I guess I could replace 'tail' in the enum case as well.